stdlib/public/runtime/RuntimeInvocationsTracking.cpp

```diff
@@ -0,0 +1,231 @@
+            object, RT_FUNCTION_ID(RT_FUNCTION));                              \
+        setGlobalRuntimeFunctionCountersMode(oldGlobalMode);                   \
+        setPerObjectRuntimeFunctionCountersMode(oldPerObjectMode);             \
+      }                                                                        \
+    }                                                                          \
+    /* Update per object counters. */                                          \
+    if (UpdatePerObjectRuntimeFunctionCounters && object) {                    \
+      RuntimeObjectStateCache[object]                                          \
+          .SWIFT_RT_FUNCTION_INVOCATION_COUNTER_NAME(RT_FUNCTION)++;           \
+      /* TODO: Remember the order/history of operations? */                    \
+    }                                                                          \
+  }
+
+#include "RuntimeInvocationsTracking.def"
+
+/// Public APIs
+
+/// Get the runtime object state associated with an object.
+SWIFT_RT_ENTRY_VISIBILITY
+void getObjectRuntimeFunctionCounters(HeapObject *object,
+                                      RuntimeFunctionCountersState *result) {
+  *result = RuntimeObjectStateCache[object];
+}
+
+/// Set the runtime object state associated with an object from a provided
+/// state.
+SWIFT_RT_ENTRY_VISIBILITY
+void setObjectRuntimeFunctionCounters(HeapObject *object,
+                                      RuntimeFunctionCountersState *state) {
+  RuntimeObjectStateCache[object] = *state;
+}
+
+/// Get the global runtime state containing the total numbers of invocations for
+/// each runtime function of interest.
+SWIFT_RT_ENTRY_VISIBILITY
+void getGlobalRuntimeFunctionCounters(RuntimeFunctionCountersState *result) {
+  *result = RuntimeGlobalFunctionCountersState;
+}
+
+/// Set the global runtime state of function pointers from a provided state.
+SWIFT_RT_ENTRY_VISIBILITY
+void setGlobalRuntimeFunctionCounters(RuntimeFunctionCountersState *state) {
+  RuntimeGlobalFunctionCountersState = *state;
+}
+
+/// Return the names of the runtime functions being tracked.
+/// Their order is the same as the order of the counters in the
+/// RuntimeObjectState structure.
+SWIFT_RT_ENTRY_VISIBILITY
+const char **getRuntimeFunctionNames() {
+  return RuntimeFunctionNames;
+}
+
+/// Return the offsets of the runtime function counters being tracked.
+/// Their order is the same as the order of the counters in the
+/// RuntimeObjectState structure.
+SWIFT_RT_ENTRY_VISIBILITY
+const uint16_t *getRuntimeFunctionCountersOffsets() {
+  return RuntimeFunctionCountersOffsets;
+}
+
+/// Return the number of runtime functions being tracked.
+SWIFT_RT_ENTRY_VISIBILITY
+uint64_t getNumRuntimeFunctionCounters() {
+  return ID_LastRuntimeFunctionName;
+}
+
+/// Define how to dump the counter for a given runtime function.
+#define FUNCTION_TO_TRACK(RT_FUNCTION)                                         \
```

Comment:
We generally put this /right/ next to the `#include`.

So I would put it like so:

```
static void dumpRuntimeCounters(...) {
uint32_t tmp = 0;
#define FUNCTION_TO_TRACK(RT_FUNCTION) \
   ... \
   ...
#include "RuntimeInvocationsTracking.def"
}
```